### DISCUSSION TITLE
Disable Prettier in JS/TS

### DISCUSSION THREAD

**Discussion by spaceunifyfifty** [zed-industries/zed#11904] [Feature Requests]
Cmd + S always auto formats my file. Here are my settings
```json
{
  "language_overrides": {
    "languages": {
      "JavaScript": {
        "format_on_save": "off"
      },
      "TypeScript": {
        "format_on_save": "off"
      }
    }
  }
}
```
How do I turn off prettier? I want eslint instead, but at this point with I will settle with just disabling formatting altogether.

**Comment by SomeoneToIgnore** [comment]
When Zed 0.137 is released next week, it will be possible to use language settings to disable prettier due to https://github.com/zed-industries/zed/pull/11558, e.g.
```
 "TypeScript": {
      "prettier": {
        "allowed": true
      }
    },
```

meanwhile, it's possible to use `"formatter": "language_server" settings to force the language server formatting always, excluding prettier from this list.

**Reply by KaKi87** [reply]
Hi,

Despite disabling Prettier in settings, a space appears by itself after a JS bracket-less `if` statement when ending the next line with a semicolon (without even saving) :

![](https://github.com/user-attachments/assets/571781a1-8532-4a46-b9a6-9114e0eb707b)

I assume there are more things like that which could occur.

How to disable these ?

Thanks

**Reply by SomeoneToIgnore** [reply]
Could you try `"use_on_type_format": false,` ?

**Reply by KaKi87** [reply]
That works, thanks !

I would like to suggest documenting this on [Configuring Zed](https://zed.dev/docs/configuring-zed).

Thanks

**Reply by thraizz** [reply]
It is also not documented in https://zed.dev/docs/languages/javascript, which has its own section about Prettier.

**Comment by spaceunifyfifty** [comment]
Is there yet a setting to do this easily?

**Reply by SomeoneToIgnore** [reply]
https://github.com/zed-industries/zed/discussions/11904#discussioncomment-9457583

**Reply by spaceunifyfifty** [reply]
@SomeoneToIgnore nothing is working, here are my settings 
```json
{
    "tab_size": 4,
    "ui_font_size": 16,
    "buffer_font_size": 20,
    "language_overrides": {
        "JavaScript": {
            "format_on_save": "off",
            "prettier": {
                "allowed": false
            }
        },
        "TypeScript": {
            "format_on_save": "off",
            "prettier": {
                "allowed": false
            }
        }
    }
}
```
`cmd + s` continues to turn single quotes into double quotes and add semi-colons to end of lines.

**Reply by spaceunifyfifty** [reply]
OK, apparently `language_overrides` is deprecated and changing it to 
```json
    "languages": {
        "JavaScript": {
            "format_on_save": "off"
        }
  ```
  fixes it. Would be great if Zed could make invalid json entries treated as errors as a way to let me know when I am typing unrecognized fields.

**Reply by SomeoneToIgnore** [reply]
Good point, we were supposed to report json schema errors but that seems to be broken somehow, sorry.

We've had an announcement about the deprecated fields both in the release notes and in https://github.com/zed-industries/zed/issues/13171 so please double check that there's no more in your config.

**Reply by dushmanta05** [reply]
> OK, apparently `language_overrides` is deprecated and changing it to
> 
> ```json
>     "languages": {
>         "JavaScript": {
>             "format_on_save": "off"
>         }
> ```
> 
> fixes it. Would be great if Zed could make invalid json entries treated as errors as a way to let me know when I am typing unrecognized fields.

Worked for turning off auto format when saved. Thank you

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
